rpgJavaInterpreter-core/src/main/kotlin/com/smeup/rpgparser/db/sql/DBSQLFile.kt

```diff
@@ -31,29 +32,45 @@ class DBSQLFile(private val name: String, private val connection: Connection) :
     }
 
     override fun readEqual(key: Value): Record {
-        if (resultSet == null) {
-            chain(emptyList())
+        return readEqual(toFields(key))
+    }
+
+    private fun filterRecord(result: Record): Record {
+        return if (result.matches(lastKey)) {
+            result
+        } else {
+            resultSet?.last()
+            Record()
         }
-        return readFromPositionedResultSet()
     }
 
     override fun readEqual(keys: List<Field>): Record {
-        TODO("not implemented")
+        val result = if (resultSet == null) {
+            chain(emptyList())
+        } else {
+            readFromPositionedResultSet()
+        }
+        lastKey = keys
+        return filterRecord(result)
     }
 
     override fun eof(): Boolean = resultSet?.isLast ?: false
 
     override fun chain(key: Value): Record {
+        return chain(toFields(key))
```

Comment:
if the readEqual and chain methods taking a Value always delegate to the ones taking a list of fields we could move them into abstract classes or implement them directly into `DBFile` as they would not be specific to this class

Reply:
We can't put it in DBFile, since this interface could (I hope) have different implementation in the future, for example with a NOSQL DB. 
IMHO an abstract class coluld be overkill here